nav2_bt_navigator/src/bt_navigator.cpp

```diff
@@ -168,7 +168,19 @@ BtNavigator::navigateToPose()
 {
   initializeGoalPose();
 
-  auto is_canceling = [this]() {return action_server_->is_cancel_requested();};
+  auto is_canceling = [this]() {
+      if (action_server_ == nullptr) {
+        RCLCPP_DEBUG(get_logger(), "Action server unavailable. Canceling.");
+        return true;
+      }
+
+      if (!action_server_->is_server_active()) {
```

Comment:
Same here, is there a reason we need to check if the action server is active on each node tick?

Reply:
Yes, since we are deactivating the action server during the call to deactivate, we need to check here in order to halt the nodes.